[DM-4313] Added documentation for services commands. (#2966)
SoftwareAG/c8y-docs: content/device-integration/fragment-library-bundle/services.md

`200,c8y_Memory,allocated,100,MB`

Similarly to measurements, alarms and events associated with the service can also be sent.

> ### Service Commands {#service-commands}

Reviewer: ```suggestion
### Service commands {#service-commands}
```

A service can announce its capability to receive commands by adding the ```c8y_ServiceCommand``` operation in its ```c8y_SupportedOperations```.
This allows the service to execute predefined or custom commands sent from the platform.

> #### Available Service Commands

Reviewer: ```suggestion
#### Available service commands
```

> The list of commands a service supports is defined in the **c8y_SupportedServiceCommands** fragment. If this fragment is not present,

Reviewer: ```suggestion
The list of commands a service supports is defined in the `c8y_SupportedServiceCommands` fragment. If this fragment is not present,
```

> the system assumes a default set of commands: **START, STOP, and RESTART**.

Reviewer: ```suggestion
the system assumes a default set of commands: `START`, `STOP`, and `RESTART`.
```

```json
{
  "c8y_SupportedServiceCommands": [
    "START",
    "STOP",
    "SAVE_SNAPSHOT"
  ]
}
```

| Name                          | Type   | Mandatory | Description |
|--------------------------------|--------|-----------|-------------|

> | `c8y_SupportedOperations`    | Array  | Yes       | Must include `"c8y_ServiceCommand"` to enable command support. |

Reviewer: ```suggestion
| `c8y_SupportedOperations`    | Array  | Yes       | Must include `"c8y_ServiceCommand` to enable command support. |
```

> | `c8y_SupportedOperations`    | Array  | Yes       | Must include `"c8y_ServiceCommand"` to enable command support. |

Reviewer: ```suggestion
| `c8y_SupportedOperations`    | Array  | Yes       | Must include `c8y_ServiceCommand` to enable command support. |
```

> | `c8y_SupportedOperations`    | Array  | Yes       | Must include `c8y_ServiceCommand` to enable command support. |
> | `c8y_SupportedServiceCommands` | Array  | No        | List of available commands for the service. If not provided, defaults to `START`, `STOP`, `RESTART`. |

Reviewer 2: ```suggestion
| `c8y_SupportedServiceCommands` | Array | No | List of available commands for the service. If not provided, defaults to `START`, `STOP`, `RESTART`. |
```
I would remove c8y_SupportedOperations from this table to stay consistent with the way its is done in most other places.

> #### Executing Service Commands

Reviewer: ```suggestion
#### Executing service commands
```

> When a command is sent, the following **operation structure** is created:

Reviewer: ```suggestion
When a command is sent, the following operation structure is created:
```

```json
{
  "deviceId": "<serviceManagedObjectId>",
  "c8y_ServiceCommand": {
    "command": "START",
    "serviceName": "Nginx Web Server",
    "serviceType": "systemd"
  }
}
```

> The operation is then processed by the device, executing the command and updating its status accordingly.

Reviewer 2: ```suggestion
The device is expected to perform the following actions:

1. Set operation status to EXECUTING
2. Find the service based in the deviceId, the serviceName, and the serviceType included in the operation
3. Execute the service command with the found service
4. Set operation status to SUCCESSFUL
```

> The operation is then processed by the device, executing the command and updating its **status** accordingly.

Reviewer: ```suggestion
The operation is then processed by the device, executing the command and updating its status accordingly.
```
